src/java/arjdbc/jdbc/RubyJdbcConnection.java

```diff
@@ -1791,7 +1791,7 @@ public static IRubyObject setRawDateTime(final IRubyObject self, final IRubyObje
     protected static IRubyObject typeCastFromDatabase(final ThreadContext context,
         final IRubyObject adapter, final RubySymbol typeName, final RubyString value) {
         final IRubyObject type = adapter.callMethod(context, "lookup_cast_type", typeName);
-        return type.callMethod(context, "type_cast_from_database", value);
+        return type.callMethod(context, "deserialize", value);
```

Comment:
Yeah, ActiveRecord now uses `deserialize` on the type and removed the `type_cast_from_database` method. It seems like this method is only called in certain cases, so I can see why you potentially wouldn't have seen it. There were only a handful of tests that were broken because of it in the postgresql adapter.